code/modules/mob/living/roguetownprocs.dm

```diff
@@ -15,45 +15,112 @@
+		BODY_ZONE_PRECISE_L_HAND = -25, // In the middle of combat? The Hands and Feet are fast and /incredibly/ unpredictable, you could land a few solid hits if you're skilled, but they're not a big, easy target at all.
+		BODY_ZONE_PRECISE_R_HAND = -25, // See above
+		BODY_ZONE_PRECISE_L_FOOT = -30, // See above above
+		BODY_ZONE_PRECISE_R_FOOT = -30, // See above above above
+
+		// General Body Zones
+		BODY_ZONE_HEAD = -20, // Head is a comparatively large target compared to most of the precision zones, but still mobile.
+		BODY_ZONE_CHEST = 0,
+		BODY_ZONE_L_ARM = -10, // Larger target but similarly fast and unpredictable like the attaches hands and feet.
+		BODY_ZONE_R_ARM = -10, // See above
+		BODY_ZONE_L_LEG = -15, // See above
+		BODY_ZONE_R_LEG = -15 // See above
+	)
+
+	// Makes the zone accuracy var apply
+	if(zone in acc_zone_penalties)
+		chance2hit += acc_zone_penalties[zone]
+
+	// Gives shorter weapons accuracy buffs and larger, more unwieldy weapons an accuracy debuff. Decapitating someone in the middle of combat in one clean swipe is not easy.
 	if(I)
-		if(I.wlength == WLENGTH_SHORT)
-			chance2hit += 10
-
-		chance2hit += ((user.STAPER-10)*5)
-
-
+		switch(I.wlength)
+			if(WLENGTH_SHORT) // Daggers and such, easy to control.
+				chance2hit += 10
+			if(WLENGTH_NORMAL) // Standard One-Handed Weapons, good control.
+				chance2hit += 5
+			if(WLENGTH_LONG) // Two-Handed weapons, decent control.
+				chance2hit += 0
+			if(WLENGTH_GREAT) // Long/Large weapons, harder to control.
+				chance2hit -= 5
+
+	// Stabbing is much more precise generally speaking
+	// Cutting is less precise but still more precise than blunt weapons
+	// When using a blunt weapon, you utilize the inertia of the swing, which makes it harder to redirect your attack on a moving/dodging target
+	if(used_intent)
+		switch(used_intent.blade_class)
+			if(BCLASS_STAB)
+				chance2hit += 8 // Stabbing is precise
+			if(BCLASS_CUT)
+				chance2hit += 4 // Cutting is more precise than a big inertia based swing
+			if(BCLASS_BLUNT)
+				chance2hit -= 4 // Once again, most of the force with blunt weapons comes from the swing, if your target moves, it's harder to adjust.
+								// I might eventually include something like this if I end up making a dodge/parry rework/overhaul for that system.
+
+	// Reworks the PER bonus that I felt either wasn't particularly impactful or ended up too overscaled with this system.
 	if(istype(user.rmb_intent, /datum/rmb_intent/aimed))
-		chance2hit += (user.STAPER)*2
+		chance2hit += round(max((user.STAPER * 1.75) - (0.05 * (user.STAPER ^ 2)), 0))
+	else
+		chance2hit += round(max((user.STAPER * 1.2) - (0.03 * (user.STAPER ^ 2)), 0))
 	if(istype(user.rmb_intent, /datum/rmb_intent/swift))
 		chance2hit -= 20
 
-	chance2hit = CLAMP(chance2hit, 5, 99)
+	// Clamped C2H so we don't get 115% or -50% rolls.
+	chance2hit = CLAMP(chance2hit, 0, 100)
```

Comment:
If I recall correctly clamp between 5 and 99 should yield a result between those numbers.
That would mean this changes it so you again can end up with zero chance to it, or 100% chance to hit.
Would need testing to confirm but yeah, clamp 5 to 95 instead is better if I remember correctly.